services/bundle_analysis/new_notify/__init__.py

```diff
@@ -0,0 +1,164 @@
+        notification_type: NotificationType,
+    ) -> NotificationFullContext | None:
+        """Builds the NotificationContext for the given notification_type
+        If the NotificationContext failed to build we can't send this notification.
+
+        Each NotificationType is paired with a ContextBuilder and MessageStrategyInterface.
+        The MessageStrategy is later used to build and send the message based on the NotificationContext
+        """
+        builders_lookup: dict[
+            NotificationType,
+            tuple[NotificationContextBuilder, MessageStrategyInterface],
+        ] = {
+            NotificationType.PR_COMMENT: (
+                BundleAnalysisPRCommentContextBuilder,
+                BundleAnalysisCommentMarkdownStrategy,
+            )
+        }
+        builder_class, message_strategy_class = builders_lookup.get(
+            notification_type, (None, None)
+        )
+
+        if builder_class is None:
+            msg = f"No context builder for {notification_type.name}. Skipping"
+            log.error(msg)
+            return None
+        try:
+            builder = builder_class().initialize_from_context(base_context)
+            return NotificationFullContext(
+                builder.build_context().get_result(),
+                message_strategy_class(),
+            )
+        except NotificationContextBuildError as exp:
+            log.error(
+                "Failed to build NotificationContext",
+                extra=dict(
+                    notification_type=notification_type, failed_step=exp.failed_step
+                ),
+            )
+            return None
+
+    def notify(self) -> BundleAnalysisNotifyReturn:
+        """Entrypoint for BundleAnalysis notifications. This function does the following:
+            1. Gets the configured notifications. Those are the ones we must send;
+            2. Attempts to build a BaseContext with necessary info for all notifications;
+            3. Attempts to build a Context for each notification to be sent;
+            4. For each notification with a context, build and send the message.
+
+        Returns: BundleAnalysisNotifyReturn - tuple with notifications configured and the ones
+            that we successfully notified.
+        """
+        notification_types = get_notification_types_configured(
+            self.current_yaml, self.owner
+        )
+        base_context = self.build_base_context()
+        if base_context is None:
+            log.warning("Skipping ALL notifications because there's no base context")
+            return BundleAnalysisNotifyReturn(
+                notifications_configured=notification_types,
+                notifications_successful=tuple(),
+            )
+
+        notification_full_contexts = filter(
+            None,
+            map(
+                partial(self.create_context_for_notification, base_context),
+                notification_types,
+            ),
+        )
```

Comment:
Similar to the comment above, if this will never be extended beyond `NotificationType.PR_COMMENT` in the future, you can probably just filter for that type early on and simplify this code quite a bit.